src/tensorlake/function_executor/service.py

```diff
@@ -256,3 +284,165 @@ def get_info(
         self, request: InfoRequest, context: grpc.ServicerContext
     ) -> InfoResponse:
         return InfoResponse(**info_response_kv_args())
+
+    def _execute_task_in_thread(
+        self, run_request: RunTaskRequest, task_execution: _TaskExecution
+    ):
+        response = RunTaskResponse(
+            task_id=run_request.task_id,
+            outcome_code=TaskOutcomeCode.TASK_OUTCOME_CODE_FAILURE,
+            failure_reason=TaskFailureReason.TASK_FAILURE_REASON_INTERNAL_ERROR,
+            stderr="",
+            stdout="",
+        )
+
+        try:
+
+            def progress_reporter(progress: Progress):
+                with task_execution.updated:
+                    task_execution.progress = ProgressUpdate(
+                        current=progress.current, total=progress.total
+                    )
+                    task_execution.updated.notify_all()
+                # sleep(0) here momentarily releases the GIL, giving
+                # other threads a chance to run.
+                time.sleep(0)
+
+                # TODO: Consider implementing task cancellation here.
+
+            # Run the task using existing handler
+            response = RunTaskHandler(
+                request=run_request,
+                invocation_state=ProxiedInvocationState(
+                    run_request.task_id, self._invocation_state_proxy_server
+                ),
+                function_wrapper=self._function_wrapper,
+                function_stdout=self._function_stdout,
+                function_stderr=self._function_stderr,
+                graph_metadata=self._graph_metadata,
+                logger=self._logger,
+                progress_reporter=progress_reporter,
+            ).run()
+
+            if response.outcome_code == TaskOutcomeCode.TASK_OUTCOME_CODE_UNKNOWN:
+                response.outcome_code = TaskOutcomeCode.TASK_OUTCOME_CODE_FAILURE
+                response.failure_reason = (
+                    TaskFailureReason.TASK_FAILURE_REASON_INTERNAL_ERROR
+                )
+
+        except Exception as e:
+            # Handle any errors by creating a failed task result
+            self._logger.error(
```

Comment:
This method _execute_task_in_thread needs to create its own local variable `logger` and use it, see how run task handler does this here https://github.com/tensorlakeai/tensorlake/blob/main/src/tensorlake/function_executor/handlers/run_function/handler.py#L33

Reply:
Why?

I've actually been meaning to ask about this - in other Python projects I've never done this passing around loggers thing; it's always been "Create a file-level logger and use it everywhere" -- i.e. what the Python logging package says is idiomatic logger usage.  But maybe the idiomatic approach isn't right for our codebase...? :-). So, what's the rationale?

Reply:
Because the caller passes a logger bound with the caller context and then the callee adds any extra callee specific context to the logger so there's no duplication or assumptions by callee about what's in the context and the context is always complete for each code path.

We even started to do the same on the Server side but it takes a while to migrate to this approach.

Reply:
Makes sense - done